bb8/src/internals.rs

```diff
@@ -41,12 +62,19 @@ where
         let conn = locked.conns.pop_front().map(|idle| idle.conn);
         let approvals = match &conn {
             Some(_) => locked.wanted(&self.statics),
-            None => locked.approvals(&self.statics, 1),
+            None => {
+                let approvals = min(1, locked.inflight_gets.saturating_sub(locked.pending_conns));
+                locked.approvals(&self.statics, approvals)
```

Comment:
I don't find this logic very clear. Maybe something like this?

```rust
let approvals = match locked.inflight_gets < locked.pending_conns {
     true => 1,
     false => 0,
};
```

(Hope I didn't get the condition inverted there.)

Reply:
Yeah that definitely reads better, thanks!